dlib/image_loader/jxl_loader.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2024  Davis E. King (davis@dlib.net), Adrià Arrufat
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_JXL_LOADER_CPp_
+#define DLIB_JXL_LOADER_CPp_
+
+// only do anything with this file if DLIB_JXL_SUPPORT is defined
+#ifdef DLIB_JXL_SUPPORT
+#include "jxl_loader.h"
+#include <jxl/decode_cxx.h>
+#include <jxl/resizable_parallel_runner_cxx.h>
+#include <fstream>
+
+namespace dlib
+{
+
+    static std::vector<unsigned char> load_contents(const std::string& filename)
+    {
+        std::ifstream stream(filename, std::ios::binary);
+        stream.exceptions(std::ifstream::failbit | std::ifstream::badbit | std::ifstream::eofbit);
+        stream.seekg(0, std::ios_base::end);
+        std::vector<unsigned char> buffer(stream.tellg());
+        stream.seekg(0);
+        stream.read(reinterpret_cast<char*>(buffer.data()), buffer.size());
```

Comment:
```suggestion
        std::vector<unsigned char> buffer;
        vectorstream temp(buffer);
        temp << stream.rdbuf();
```
This might be faster.  Used to be the fastest way to do this kind of thing with streams.  It's more readable in any case :)